### ISSUE TITLE
Make reset mappings in between tests configurable so the wiremock server port is retained

### ISSUE THREAD

**Issue by jonasgeiregat** [spring-cloud/spring-cloud-contract#1791]
I'm unable to reuse the dynamically assigned port in between multiple tests runs. 
Given I have 2 tests classes annotated with `@AutoConfigureWireMock(port = 0)`
And the spring application receives the dynamically assigned port through usage of the replaced`${wiremock.server.port}` property. In between the first and second test the port has changed and the 2nd test always fails being unable to reach the server because it's not running on the first assigned port but a new one.

**Describe the solution you'd like**
`WireMockTestExecutionListener` is responsible for this behavior and it could be made configurable to not reset mappings after test execution, hence not restarting the server. 
`wirenock.reset-mappings=false` property could be introduced
Alternatively or additionally a property on the `@AutoConfigureWireMock` annotation could also be added.

**Comment by marcingrzejszczak** [comment]
Do you have a small sample that replicates the problem? Which version of SCC is this problem related to?

**Comment by jonasgeiregat** [comment]
I tried to replicate it in a separate project but to my surprise was unable to.

I went back to my actual project and found out that when I put a breakpoint in `WireMockConfiguration` where it starts the server for the second time and wait a bit,  both tests pass. The same behavior can be noticed when I simply put a sleep right at the beginning of my test. 

It might also be worth noting that the tests are written using groovy and the spock framework.
